src/OpenSpaceToolkit/Astrodynamics/Trajectory/Models/Tabulated.cpp

```diff
@@ -103,39 +169,19 @@ State                           Tabulated::calculateStateAt                 (
         throw ostk::core::error::runtime::Undefined("Tabulated") ;
     }
 
-    const Pair<const State*, const State*> stateRange = this->accessStateRangeAt(anInstant) ;
-
-    if ((stateRange.first != nullptr) && (stateRange.second != nullptr))
+    if (anInstant < firstState_.accessInstant() || anInstant > lastState_.accessInstant())
     {
+        throw ostk::core::error::RuntimeError(String::Format("Provided instant [{}] is outside of interpolation range [{}, {}].", anInstant.toString(), firstState_.accessInstant().toString(), lastState_.accessInstant().toString())) ;
+    }
 
-        const State& previousState = *(stateRange.first) ;
-        const State& nextState = *(stateRange.second) ;
-
-        const Real ratio = Duration::Between(previousState.accessInstant(), anInstant).inSeconds() / Duration::Between(previousState.accessInstant(), nextState.accessInstant()).inSeconds() ;
-
-        const Vector3d x_ECI = previousState.accessPosition().accessCoordinates() + ratio * (nextState.accessPosition().accessCoordinates() - previousState.accessPosition().accessCoordinates()) ;
-        const Vector3d v_ECI = previousState.accessVelocity().accessCoordinates() + ratio * (nextState.accessVelocity().accessCoordinates() - previousState.accessVelocity().accessCoordinates()) ;
-
-        const Position position = { x_ECI, previousState.accessPosition().getUnit(), previousState.accessPosition().accessFrame() } ;
-        const Velocity velocity = { v_ECI, previousState.accessVelocity().getUnit(), previousState.accessVelocity().accessFrame() } ;
-
-        const State state = { anInstant, position, velocity } ;
-
-        return state ;
+    VectorXd interpolatedCoordinates(interpolators_.getSize()) ;
 
-    }
-    else if (stateRange.first != nullptr)
+    for (Size i = 0 ; i < interpolators_.getSize() ; ++i)
```

Comment:
Size > Index (that's a nit for proper understanding because IIRC the `typedef` points to the same implementation under the hood)